components/menu/menu-item-link.js

```diff
@@ -129,19 +136,27 @@ class MenuItemLink extends (newWindowIconEnabled ? LinkMixin(MenuItemMixin(LitEl
 		return null;
 	}
 
+	// remove this function when cleaning up GAUD-8369-menu-item-link-click-changes
 	_onClick() {
 		if (this.shadowRoot) this.shadowRoot.querySelector('a').dispatchEvent(new CustomEvent('click'));
 	}
 
 	_onKeyDown(e) {
-		if (e.keyCode === this.__keyCodes.ENTER || e.keyCode === this.__keyCodes.SPACE) {
-			const target = this._getTarget();
-			if (target === '_parent') {
-				window.parent.location.assign(this.href);
-			} else if (target === '_top') {
-				window.top.location.assign(this.href);
-			} else {
-				window.location.assign(this.href);
```

Comment:
From what I can tell, this is all just handled for us if we `click` (which we want to do for the router anyways).